components/script/dom/filereadersync.rs

```diff
@@ -2,23 +2,37 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use dom::bindings::codegen::Bindings::FileReaderSyncBinding;
-use dom::bindings::error::Fallible;
+use base64;
+use dom::bindings::codegen::Bindings::BlobBinding::BlobMethods;
+use dom::bindings::codegen::Bindings::FileReaderSyncBinding::{FileReaderSyncBinding, FileReaderSyncMethods};
+use dom::bindings::error::{Error, Fallible};
+use dom::bindings::nonnull::NonNullJSObjectPtr;
 use dom::bindings::reflector::reflect_dom_object;
 use dom::bindings::root::DomRoot;
+use dom::bindings::str::DOMString;
+use dom::blob::Blob;
 use dom::eventtarget::EventTarget;
+use dom::filereader::FileReaderReadyState;
 use dom::globalscope::GlobalScope;
 use dom_struct::dom_struct;
+use encoding_rs::{Encoding, UTF_8};
+use hyper::mime::{Attr, Mime};
+use js::jsapi::JSContext;
+use js::typedarray::{ArrayBuffer, CreateWith};
+use std::cell::Cell;
+use std::ptr;
 
 #[dom_struct]
 pub struct FileReaderSync {
-    eventtarget: EventTarget
+    eventtarget: EventTarget,
+    ready_state: Cell<FileReaderReadyState>,
 }
 
 impl FileReaderSync {
     pub fn new_inherited() -> FileReaderSync {
         FileReaderSync {
             eventtarget: EventTarget::new_inherited(),
+            ready_state: Cell::new(FileReaderReadyState::Empty),
```

Comment:
Since the spec doesn't actually tell us to modify the readyState for synchronous file readers, nor does it describe a way where the LOADING check could actually succeed, let's remove it from our implementation.

Reply:
I've filed https://github.com/w3c/FileAPI/issues/96 against the spec.